src/optimizer/rule/RemoveUselessProjectRule.cpp

```diff
@@ -0,0 +1,82 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "optimizer/rule/RemoveUselessProjectRule.h"
+
+#include "optimizer/OptContext.h"
+#include "optimizer/OptGroup.h"
+#include "planner/PlanNode.h"
+#include "planner/Query.h"
+
+using nebula::graph::PlanNode;
+using nebula::graph::QueryContext;
+
+namespace nebula {
+namespace opt {
+
+std::unique_ptr<OptRule> RemoveUselessProjectRule::kInstance =
+    std::unique_ptr<RemoveUselessProjectRule>(new RemoveUselessProjectRule());
+
+RemoveUselessProjectRule::RemoveUselessProjectRule() {
+    RuleSet::QueryRules().addRule(this);
+}
+
+const Pattern& RemoveUselessProjectRule::pattern() const {
+    static Pattern pattern = Pattern::create(graph::PlanNode::Kind::kProject,
+                                             {Pattern::create(graph::PlanNode::Kind::kAggregate)});
+    return pattern;
+}
+
+StatusOr<OptRule::TransformResult> RemoveUselessProjectRule::transform(
```

Comment:
I advise you to rename the rule since the pattern of this rule is `Project-Aggregate` not general case. make it clear in name.

Reply:
ACK.